MobiusCore/Source/EffectHandlers/EffectPath.swift

```diff
@@ -0,0 +1,27 @@
+
+// Copyright (c) 2019 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+public struct EffectPath<Input, Output> {
```

Comment:
Does this type need to be public? We can explore it later, I guess, but perhaps it would be better to have some kind of anonymous functions in the public API. Not sure if that makes things easier or harder to learn.

Reply:
Yes, let's explore it later. I think the DSL like things should be able to replace it. I'm not sure if that's the right abstraction.

Reply:
Another name might be something along the lines of 'Gate', as it's something that is conditionally open.